File: day04/day04.rb
input = File.read('input.txt').chomp.split("\n").map(&:split)
puts input.count{ |line| line.uniq == line }
puts input.count{ |line| line.map!(&:chars).map!(&:sort).uniq == line }

# puts input.each_line.inject(0) { |sum, line| line.split.uniq! ? sum : sum + 1 }
# puts input.each_line.inject(0) { |sum, line| line.split.map!(&:chars).map!(&:sort).uniq! ? sum : sum + 1 }

# puts File.read('input.txt').chomp.split("\n").map(&:split).count{ |line| !line.uniq! }
# puts input.each_line.select{ |line| !line.split.uniq! }.lengths
# puts input.each_line.count(&:uniq)
# puts input.each_line.select{ |line| !line.split.map!(&:chars).map!(&:sort).uniq! }.length

File: day04/day04.exs
defmodule Day04 do
  def no_dups?(sentence) do
    words = String.split(sentence)
    words == Enum.uniq(words)
  end
end

File.read!("input.txt")
|> String.split("\n", trim: true)
|> Enum.filter(&Day04.no_dups?/1)
|> length
|> IO.puts

File.read!("input.txt")
|> String.split("\n", trim: true)
|> Enum.filter(&Day04.no_dups?/1)
|> length
|> IO.puts

# File.read!("input.txt")
# |> String.trim
# |> String.split("\n")
# |> Enum.reduce(0, &Day04.check_dups/2)
# |> IO.puts

# "here is a test"
# |> String.split
# |> Enum.sort
# |> IO.inspect

# "testing"
# |> String.graphemes
# |> Enum.sort
# |> IO.inspect

